client/doc_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"go.opentelemetry.io/collector/client"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+func Example_receiver() {
```

Comment:
This example doesn't seem to match the description of how things work, which says:
>Receivers are responsible for obtaining a client.Info from the current context and enhancing the client.Info
with the net.Addr from the peer

In this example the receiver is not obtaining the client.Info, it creates the client.Info. I think it is worth clarifying a bit more the "create" vs "obtain" part in the description.

Reply:
Not sure I follow. The example has this:

```go
	// Get the client from the context: if it doesn't exist, FromContext will
	// create one
	cl := client.FromContext(ctx)
```

This is obtaining the client.Info from the context, it's not creating one (not itself, but perhaps client.FromContext will).